### PR TITLE
Issue 30 robot mesh

### PR DESCRIPTION
**Pull request by rubenlucas93**
robot mesh implementation and some refactors to let rl-studio work with different problems

### CODE REVIEW

File: rl_studio/agents/f1/manual_pilot.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 import gym
 
-from rl_studio.agents.f1 import settings
+from agents.f1 import settings
```
Comment (igarag) [inline rl_studio/agents/f1/manual_pilot.py:9]: Did you have any problems with this import? I think it reads the library dependencies better if we start from the root, don't you think?

File: rl_studio/agents/f1/train_dqn.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
 import matplotlib.pyplot as plt
 import numpy as np
 from keras import backend as K
-from settings import my_board
+from agents.f1.settings import my_board
```
Comment (igarag) [inline rl_studio/agents/f1/train_dqn.py:16]: Same here

File: rl_studio/agents/f1/train_qlearn.py [hunk 3]
```diff
@@ -6,18 +6,19 @@
 import gym
 import numpy as np
 
-from agents import utils, liveplot
+from agents import liveplot
+from agents.f1 import utils
 from algorithms.qlearn import QLearn
-from rl_studio.agents.f1.settings import QLearnConfig
+from agents.f1.settings import QLearnConfig
```
Comment (igarag) [inline rl_studio/agents/f1/train_qlearn.py:12]: Same here with those imports.

File: rl_studio/agents/f1/utils.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    # STATES CUMULATED REWARD
+    states_cum_reward_file_name = base_file_name + "_STATES_CUM_REWARD.pkl"
+    file_dump = open(
+        "./logs/qlearn_models/3_" + current_time + states_cum_reward_file_name, "wb"
+    )
+    pickle.dump(states_rewards, file_dump)
+    # STATES
+    steps = base_file_name + "_STATES_STEPS.pkl"
+    file_dump = open("./logs/qlearn_models/4_" + current_time + steps, "wb")
+    pickle.dump(states, file_dump)
```
Comment (igarag) [inline rl_studio/agents/f1/utils.py:57]: I  think we can change this strings by f-strings:

```python
 file_dump = open(
      "./logs/qlearn_models/3_" + current_time + states_cum_reward_file_name, "wb"
  )
```

f-string:

```bash
 file_dump = open(f"./logs/qlearn_models/3_{current_time}{states_cum_reward_file_name},wb")
```

File: rl_studio/agents/robot_mesh/manual_pilot.py [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+# from gym.envs.registration import register
+
+# # my envs
+# register(
+#     id='mySim-v0',
+#     entry_point='envs:MyEnv',
+#     # More arguments here
+# )
+
```
Comment (igarag) [inline rl_studio/agents/robot_mesh/manual_pilot.py:20]: Does this code not belong to the registry file?

File: rl_studio/agents/robot_mesh/manual_pilot.py [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+                    env._flush(force=True)
+                    if not env.done:
+                        state = nextState
+                    else:
+                        break
+                elif inpt == "q":
+                    system.exit(1)
+                else:
+                    print("wrong action! Try again")
+
```
Comment (igarag) [inline rl_studio/agents/robot_mesh/manual_pilot.py:112]: maybe we could break this function into smaller, more manageable functions and take the opportunity to refactor this program, don't you think?

File: rl_studio/agents/robot_mesh/train_qlearn.py [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+
+                print("Selected Action!! " + str(action))
+                # Execute the action and get feedback
+                nextState, reward, done, lap_completed = env.step(action)
+                cumulated_reward += reward
+
+                if highest_reward < cumulated_reward:
+                    highest_reward = cumulated_reward
+
+                # nextState = ''.join(map(str, observation))
```
Comment (igarag) [inline rl_studio/agents/robot_mesh/train_qlearn.py:119]: The same split here.

File: rl_studio/agents/settings.py [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+class QLearnConfig(BaseModel):
+    actions: int = 3
+    debug_level: int = 0
+    telemetry: bool = False
+    telemetry_mask: bool = True
+    plotter_graphic: bool = False
+    my_board: bool = True
+    save_positions: bool = False
+    save_model: bool = True
+    load_model: bool = False
```
Comment (igarag) [inline rl_studio/agents/settings.py:16]: Maybe we could change the `BaseModel` by the `BaseSettings`. You can check this class in their documentation, [here](https://pydantic-docs.helpmanual.io/usage/settings/).

File: rl_studio/envs/__init__.py [hunk 9]
```diff
@@ -1,2 +1,2 @@
-from rl_studio.envs.gazebo_env import GazeboEnv
+from envs.gazebo_envs import GazeboEnv
```
Comment (igarag) [inline rl_studio/envs/__init__.py:1]: We must keep the root directory

File: rl_studio/envs/gazebo_envs.py [hunk 10]
```diff
@@ -0,0 +1,260 @@
...
+from pathlib import Path
+
+from gazebo_msgs.msg import ModelState
+from gazebo_msgs.srv import SetModelState, GetModelState
+
+import subprocess
+from std_srvs.srv import Empty
+import random
+from rosgraph_msgs.msg import Clock
+
```
Comment (igarag) [inline rl_studio/envs/gazebo_envs.py:17]: what is the difference between this `gazebo_envs` and the same stored inside of `envs/f1/gazebo_envs`?, maybe we could merge into one. What do you think?

**Review by igarag — CHANGES_REQUESTED** [review]
Tell me your opinion about the comments :-)

File: rl_studio/agents/f1/manual_pilot.py (hunk 1 above)
Comment (rubenlucas93) [reply rl_studio/agents/f1/manual_pilot.py:9]: right, I change it today.

File: rl_studio/agents/f1/utils.py (hunk 4 above)
Comment (rubenlucas93) [reply rl_studio/agents/f1/utils.py:57]: I didn´t know them. Thanks!!! :) I change them today

File: rl_studio/agents/robot_mesh/manual_pilot.py (hunk 5 above)
Comment (rubenlucas93) [reply rl_studio/agents/robot_mesh/manual_pilot.py:20]: Yes. I changed and test it today

File: rl_studio/agents/robot_mesh/manual_pilot.py (hunk 6 above)
Comment (rubenlucas93) [reply rl_studio/agents/robot_mesh/manual_pilot.py:112]: Totally right

File: rl_studio/envs/gazebo_envs.py (hunk 10 above)
Comment (rubenlucas93) [reply rl_studio/envs/gazebo_envs.py:17]: Good idea. Thanks!

File: rl_studio/envs/__init__.py (hunk 9 above)
Comment (igarag) [reply rl_studio/envs/__init__.py:1]: You did not change this value?

File: rl_studio/envs/f1/image_f1.py [hunk 11]
```diff
@@ -1,7 +1,7 @@
 import cv2
 import numpy as np
 
-from rl_studio.agents.f1.settings import QLearnConfig
+from agents.f1.settings import QLearnConfig
```
Comment (igarag) [inline rl_studio/envs/f1/image_f1.py:4]: Include the absolute path to this class.

Comment (rubenlucas93) [reply rl_studio/envs/f1/image_f1.py]: Done

**Review by igarag — APPROVED** [review]

**Pull request merged**